src/sema/expression.rs

```diff
@@ -6466,7 +6477,7 @@ fn method_call_named_args(
                     }
                 };
 
-                match cast(&arg.loc(), arg.clone(), &param.ty, true, ns, diagnostics) {
+                match arg.cast(&arg.loc(), &param.ty, true, ns) {
```

Comment:
again here the diagnostic should go to `diagnostics` not `ns.diagnostics`, if this particular overloaded function does not match then the loop should try the next one without failing.

This is an example of some solidity which fails on your branch, not on main:
```
contract c {
	function foo(int, bool) public { }
	function foo(bool, int) public { }
	function bar() public {
		foo(1, true);
		foo(true, 1);
	}
}
```

Reply:
The same overloaded function match exists for named method call, function call, named function call, constructor call, named constructor call.

Reply:
@LucasSte Please add this example to test cases